src/prefect/server/models/block_documents.py

```diff
@@ -260,52 +260,68 @@ async def read_block_document_by_name(
 
 
 @inject_db
-async def read_block_documents(
-    session: AsyncSession,
+def _apply_block_document_filters(
+    query,
     db: PrefectDBInterface,
     block_document_filter: Optional[schemas.filters.BlockDocumentFilter] = None,
-    block_type_filter: Optional[schemas.filters.BlockTypeFilter] = None,
-    block_schema_filter: Optional[schemas.filters.BlockSchemaFilter] = None,
-    include_secrets: bool = False,
-    sort: Optional[
-        schemas.sorting.BlockDocumentSort
-    ] = schemas.sorting.BlockDocumentSort.NAME_ASC,
-    offset: Optional[int] = None,
-    limit: Optional[int] = None,
+    block_schema_filter: Optional[schemas.filters.BlockDocumentFilter] = None,
+    block_type_filter: Optional[schemas.filters.BlockDocumentFilter] = None,
```

Comment:
nit: I think these types can be tightened to `Optional[BlockTypeFilter]` and `Optional[BlockSchemaFilter]`